plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -17,149 +18,55 @@
-           "int" and/or "log" modifiers can be added in front to cast to integer or
-           use log-distributed values (Eg: int:log:4:1024)
-         - anything else will be treated as a constant string
-       * a config definition dict for scalar parameters, with potential fields
-         init, lower, upper, step, log, integer
-       * a list for option parameters defined in config file
+    # intervals are scalars:
+    # Note: by intervals are always interpreted as floats (even for int start and end values).
+    interval(0,1)     -> RangeSweep(start=0.0, end=1.0) -> ng.p.Scalar(lower=0.0, upper=1.0)
+    interval(0.0,1.0) -> RangeSweep(start=0.0, end=1.0) -> ng.p.Scalar(lower=0.0, upper=1.0)
 
-    Returns
-    -------
-    Parameter or str
-        A Parameter if the string fitted one of the definitions, else the input string
+    # a user can cast the interval to int to override that:
+    int(interval(0,1) -> RangeSweep(start=0, end=1) -> ng.p.Scalar(lower=0.0, upper=1.0).set_integer_casting()
     """
-    # lazy initialization to avoid overhead when loading hydra
-    import nevergrad as ng
-
-    # revert config parsing
-
-    if isinstance(description, (ListConfig, list)):
-        description = ",".join(str(x) for x in description)
-    if isinstance(description, str):
-        # cast to spec if possible
-        try:
-            description = CommandlineSpec.parse(description)
-        except ValueError:
-            pass
-    # convert scalar commandline specs to dict
-    if isinstance(description, CommandlineSpec) and description.bounds is not None:
-        description = ScalarConfigSpec(
-            lower=description.bounds[0],
-            upper=description.bounds[1],
-            log=description.log,
-            integer=description.cast == "int",
-        )
-    # convert scalar config specs to dict
-    # convert dict to Scalar parameter instance
+    scalar = None
+    if isinstance(description, Override):
+        override = description
+        val = override.value()
+        if override.is_sweep_override():
+            if override.is_choice_sweep():
+                if "ordered" in val.tags:
+                    return ng.p.TransitionChoice(val.list)
+                else:
+                    return ng.p.Choice(val.list)
+            elif override.is_range_sweep():
+                params = {"lower": val.start, "upper": val.stop}
+                scalar = ng.p.Scalar(**params)
+                scalar.set_mutation(sigma=val.step)
+                scalar.integer = isinstance(val.start, int)
+            elif override.is_interval_sweep():  # continuous variable
+                if "log" in val.tags:
+                    scalar = ng.p.Log(lower=val.start, upper=val.end)
+                else:
+                    scalar = ng.p.Scalar(lower=val.start, upper=val.end)
+                if isinstance(val.start, int):
+                    scalar.set_integer_casting()
```

Comment:
@omry 

I'm getting a lot of complaints from `mypy` here, I'm wondering if you have any suggestion on how to handle it more cleanly.


```
(pytest38) jieru-mbp:hydra jieru$ mypy --strict plugins/hydra_nevergrad_sweeper/hydra_plugins --config-file .mypy.ini 
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:48: error: Name 'val' already defined on line 46
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "str" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "int" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "float" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "bool" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "List[Any]" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "Dict[str, Any]" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:49: error: Item "None" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "tags"
plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py:50: error: Item "str" of "Union[str, int, float, bool, List[Any], Dict[str, Any], ChoiceSweep, RangeSweep, IntervalSweep, None]" has no attribute "list"

```

Reply:
not sure why.

Reply:
i needed to cast the vals in the corresponding sweeps :)